pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -182,6 +182,12 @@ if [ -n "$VAULT_CLIENT_KEY" ]; then
 	ARGS+=(--key "${VAULT_CLIENT_KEY}")
 fi
 
+# For a request to any host/port, connect to VAULT_TLS_SERVER_NAME:request's original port instead
+# Used for SNI validation and correct certificate matching
+if [ -n "$VAULT_TLS_SERVER_NAME" ]; then
+	ARGS+=(--connect-to ::"${VAULT_TLS_SERVER_NAME}":)
```

Comment:
Is the last `:` intended? It seems odd to include it if there is no port number, but I guess that's the expected syntax?
```suggestion
	ARGS+=(--connect-to ::"${VAULT_TLS_SERVER_NAME}")
```

Reply:
As odd as it seems, this is the correct syntax. Also, see the man page:

> For a request to the given HOST1:PORT1 pair, connect to HOST2:PORT2 instead.  This option is suitable to direct requests at a specific server, e.g. at a specific cluster node in a cluster of servers. This option is only used to establish the network connection. It does NOT affect the hostname/port that is used for TLS/SSL (e.g. SNI, certificate verification) or for the application  protocols.
              "HOST1" and "PORT1" may be the empty string, meaning "any host/port". "HOST2" and "PORT2" may also be the empty string, meaning "use the request's original host/port".